src/TableFunctions/TableFunctionOneHotEncodingView.cpp

```diff
@@ -0,0 +1,140 @@
+#include <Parsers/ASTFunction.h>
+#include <Parsers/queryToString.h>
+#include <Parsers/parseQuery.h>
+#include <Parsers/ASTSelectWithUnionQuery.h>
+#include <Parsers/ParserSelectWithUnionQuery.h>
+#include <Interpreters/InterpreterSelectWithUnionQuery.h>
+#include <Storages/StorageView.h>
+#include <TableFunctions/ITableFunction.h>
+#include <TableFunctions/TableFunctionFactory.h>
+#include <TableFunctions/TableFunctionOneHotEncodingView.h>
+#include "registerTableFunctions.h"
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+ColumnsDescription TableFunctionOneHotEncodingView::getActualBaseQueryTableStructure(
+    ASTPtr base_query,
+    ContextPtr context)
+{
+    if (base_query)
+    {
+        auto sample = InterpreterSelectWithUnionQuery::getSampleBlock(base_query, context);
+        return ColumnsDescription(sample.getNamesAndTypesList());
+    }
+    return ColumnsDescription();
+}
+
+std::tuple<const String, const String> TableFunctionOneHotEncodingView::getWithAndSelectForColumn(
+    String base_query_str,
+    const String & column_name,
+    const size_t & column_idx)
+{
+    String column_idx_str = std::to_string(column_idx);
+    /// create column name and encoded column name place holders
+    /// because we can't use the actual names
+    /// due to potential SQL injection attacks and therefore we will
+    /// replace these in the final AST tree using visitors
+    String ohe_column_name = "__ohe_column_name_" + column_idx_str;
+    String encoded_column_name = "__ohe_encoded_column_name_" +  column_idx_str;
+
+    String mapping = "__ohe_map_" + column_idx_str;
+    String mapping_idx = "__ohe_map_idx_" + column_idx_str;
+    String with;
+    String select;
+
+    with = "(SELECT groupArray(`" + ohe_column_name +"`) FROM ";
```

Comment:
We cannot merge this code.

Reply:
> The implementation looks very dirty. Maybe take some time to reconsider this feature or look for alternative?

This feature was just an experiment. I will close this PR.

Reply:
> We cannot merge this code.

While it might not be pretty, I don't believe this code has any SQL injection issues. The choice of not defining query in AST was deliberate due to impracticality of defining complex query using an AST tree.